File: fundaments/block4-JS-introduction/day04-objects-and-functions/part2-functions/exercise4.js
// Crie uma função que receba um array de nomes e retorne o nome com a maior quantidade de caracteres.

let arrayTeste = ['José', 'Lucas', 'Nádia', 'Fernanda', 'Cairo', 'Joana'];

function biggerName(nameList) {
  let numberOfLetters = nameList[0].length
  let posicao = 0;

  for(let index = 0; index < nameList.length; index += 1){
    if (nameList[index].length > numberOfLetters){
      numberOfLetters = arrayTeste[index].length
      posicao = index;
    }
  }
  return nameList[posicao]
}

console.log(biggerName(arrayTeste));

File: computer-science/block33-python-intro/day01-learning-python/exercise4.py
# Crie uma função que receba uma lista de nomes e retorne
# o nome com a maior quantidade de caracteres.
# Por exemplo, para ["José", "Lucas", "Nádia", "Fernanda", "Cairo", "Joana"],
# o retorno deve ser "Fernanda".

exemplo = ["José", "Lucas", "Nádia", "Fernanda", "Cairo", "Joana"]


def check_names(list):
    name_length = 0
    for name in list:
        if len(name) > name_length:
            name_length = len(name)
            result = name
    return result


print(check_names(exemplo))
